utils.go

```diff
@@ -276,6 +276,55 @@ func parseBinaryDateTime(num uint64, data []byte, loc *time.Location) (driver.Va
 	return nil, fmt.Errorf("invalid DATETIME packet length %d", num)
 }
 
+func appendDateTime(buf []byte, t time.Time) ([]byte, error) {
+	nsec := t.Nanosecond()
+	// to round under microsecond
+	if nsec%1000 >= 500 { // save half of time.Time.Add calls
+		t = t.Add(500 * time.Nanosecond)
+		nsec = t.Nanosecond()
+	}
```

Comment:
Hmm, I want to remove this rounding behavior.  But it should be a separated issue.  No problem here.

Reply:
It’s no problem for me to remove it

Reply:
actually I want to remove it too